tests/slsa_analyzer/checks/resources/pypi_files/zlibxjson.html

```diff
@@ -0,0 +1,968 @@
+            <span class="reset-text margin-top">
+              <a href="/help/#project-release-notifications">Release notifications</a> |
+              <a href="/rss/project/zlibxjson/releases.xml">RSS feed <i class="fa fa-rss" aria-hidden="true"></i></a>
+            </span>
+          </h2>
+
+          <div class="release-timeline">
+            <div class="release release--latest release--current">
+              <div class="release__meta">
+                <span class="badge">This version</span>
+              </div>
+
+              <div class="release__graphic">
+                <div class="release__line"></div>
+                <img class="release__node" alt="" src="https://pypi.org/static/images/blue-cube.572a5bfb.svg">
+              </div>
+
+              <a class="card release__card" href="/project/zlibxjson/8.2/">
+                <p class="release__version">
+                  8.2
+                </p>
+                <p class="release__version-date">
+                  <time datetime="2024-06-29T20:46:26+0000" data-controller="localized-time" data-localized-time-relative="true" data-localized-time-show-time="false">
+  Jun 29, 2024
+</time>
+                </p>
+              </a>
+            </div>
+            <div class="release release--oldest">
+              <div class="release__meta">
+              </div>
+
+              <div class="release__graphic">
+                <div class="release__line"></div>
+                <img class="release__node" alt="" src="https://pypi.org/static/images/white-cube.2351a86c.svg">
+              </div>
+
+              <a class="card release__card" href="/project/zlibxjson/8.1/">
+                <p class="release__version">
+                  8.1
+                </p>
+                <p class="release__version-date">
+                  <time datetime="2024-06-29T20:22:51+0000" data-controller="localized-time" data-localized-time-relative="true" data-localized-time-show-time="false">
+  Jun 29, 2024
+</time>
+                </p>
+              </a>
+            </div>
+          </div>
+        </div>
+
+          <div id="files" data-project-tabs-target="content" class="vertical-tabs__content" role="tabpanel" aria-labelledby="files-tab mobile-files-tab" tabindex="-1">
+            <h2 class="page-title">Download files</h2>
+            <p>Download the file for your platform. If you're not sure which to choose, learn more about <a href="https://packaging.python.org/tutorials/installing-packages/" title="External link" target="_blank" rel="noopener">installing packages</a>.</p>
+            <h3>
+Source Distribution            </h3>
+
+                  <div class="file">
+      <div class="file__graphic">
+        <i class="far fa-file" aria-hidden="true"></i>
+      </div>
+
+      <div class="card file__card">
+        <a href="https://files.pythonhosted.org/packages/3e/1e/b1ecb05e7ca1eb74ca6257a7f43d052b90d2ac01feb28eb28ce677a871ab/zlibxjson-8.2.tar.gz">
```

Comment:
This is not relevant to this PR, but I have some ideas for improvements:
1. Because we rely heavily on the HTML content (which is not guarantee to be stable), we need to have an integration test to test whether our logic for obtaining and processing the HTML content is not breaking (similar to what we have for repo finder https://github.com/oracle/macaron/blob/429ba95abc6ee0b917d2115598838b07a4244caf/tests/integration/cases/repo_finder_remote_calls.
2. I don't know if this would be possible. But we can change this HTML content into a template file (using Jinja2, which is already available) where the information that we are interested in (i.e. the source URL) into template string. Then the test will provide the necessary values and generate a HTML content from that template. We can then feel free to view and parameterize those values from within the test module. 
Please let me know what you think.

Reply:
I don't agree with mocking this behavior and that's why I am using the actual HTML content to exactly test HTML content processing. Note that this HTML content is the real one and has not been modified.

Reply:
Ah I see. Thanks Behnaz.